### ISSUE TITLE
博主，为什么计算出来是很低，和你教程的过程一样的。但出来值很低，差不多一半。

### ISSUE THREAD

**Issue by lijia2019310** [percent4/ALBERT_NER_KERAS#5]

**Comment by 18140663659** [comment]
我训练出来的结果也不理想

**Comment by zhangbc** [comment]
我原封不动的跑了代码，epochs=10，loss却出现了负值。

Using TensorFlow backend.
WARNING:tensorflow:From /content/drive/My Drive/ner-project/ALBERT_NER_KERAS/albert_zh/graph.py:18: The name tf.logging.set_verbosity is deprecated. Please use tf.compat.v1.logging.set_verbosity instead.

WARNING:tensorflow:From /content/drive/My Drive/ner-project/ALBERT_NER_KERAS/albert_zh/graph.py:18: The name tf.logging.DEBUG is deprecated. Please use tf.compat.v1.logging.DEBUG instead.

WARNING:tensorflow:From /content/drive/My Drive/ner-project/ALBERT_NER_KERAS/albert_zh/graph.py:18: The name tf.logging.ERROR is deprecated. Please use tf.compat.v1.logging.ERROR instead.

I:BERT_VEC:[graph:opt: 48]:model config: /content/drive/My Drive/ner-project/ALBERT_NER_KERAS/albert_zh/albert_tiny/albert_config_tiny.json
I:BERT_VEC:[graph:opt: 54]:build graph...
embedding_lookup_factorized. factorized embedding parameterization is used.
ln_type: postln
old structure of transformer.use: transformer_model,which use post-LN
I:BERT_VEC:[graph:opt:128]:load parameters from checkpoint...
I:BERT_VEC:[graph:opt:130]:freeze...
I:BERT_VEC:[graph:opt:133]:optimize...
I:BERT_VEC:[graph:opt:144]:write graph to a tmp file: ./tmp_graph11
sentences length: 20864 
last sentence:  思想自由是对自我而言，用中国传统的说法是有所为；兼容并包是指对待他人，要有所不为。
start ALBERT encding
Processing: 100% 20864/20864 [01:05<00:00, 317.46it/s]
tcmalloc: large alloc 5207654400 bytes == 0x406ee000 @  0x7f1524f671e7 0x7f15228bb5e1 0x7f152291fc78 0x7f1522922db8 0x7f1522923395 0x7f15229ba65d 0x50a4a5 0x50beb4 0x5095c8 0x50a2fd 0x50beb4 0x5095c8 0x50a2fd 0x50beb4 0x507be4 0x50ad03 0x634e72 0x634f27 0x6386df 0x639281 0x4b0dc0 0x7f1524b64bf7 0x5b259a
end ALBERT encoding
sentences length: 2318 
last sentence:  良性肿瘤、恶性肿瘤虽然只是一字之差，但两者有根本性的差别。
start ALBERT encding
Processing: 100% 2318/2318 [00:06<00:00, 353.14it/s]
end ALBERT encoding
sentences length: 4636 
last sentence:  因此，村民进行民主选举的心态是在这样一种背景映衬下加以表现的，这无疑给该片增添了几分厚重的历史文化氛围。
start ALBERT encding
Processing: 100% 4636/4636 [00:13<00:00, 355.21it/s]
end ALBERT encoding
/usr/local/lib/python3.6/dist-packages/keras_contrib/layers/crf.py:346: UserWarning: CRF.loss_function is deprecated and it might be removed in the future. Please use losses.crf_loss instead.
  warnings.warn('CRF.loss_function is deprecated '
/usr/local/lib/python3.6/dist-packages/keras_contrib/layers/crf.py:353: UserWarning: CRF.accuracy is deprecated and it might be removed in the future. Please use metrics.crf_accuracy
  warnings.warn('CRF.accuracy is deprecated and it '
Model: "model_1"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
bert_output (InputLayer)     (None, 200, 312)          0         
_________________________________________________________________
bi_lstm (Bidirectional)      (None, 200, 256)          451584    
_________________________________________________________________
dropout (Dropout)            (None, 200, 256)          0         
_________________________________________________________________
time_distributed (TimeDistri (None, 200, 8)            2056      
_________________________________________________________________
crf_1 (CRF)                  (None, 200, 8)            152       
=================================================================
Total params: 453,792
Trainable params: 453,792
Non-trainable params: 0
_________________________________________________________________
Train on 20864 samples, validate on 2318 samples
Epoch 1/10
20864/20864 [==============================] - 571s 27ms/step - loss: 0.4005 - crf_viterbi_accuracy: 0.9608 - val_loss: 0.0945 - val_crf_viterbi_accuracy: 0.9735
Epoch 2/10
20864/20864 [==============================] - 569s 27ms/step - loss: 0.0660 - crf_viterbi_accuracy: 0.9782 - val_loss: 0.0478 - val_crf_viterbi_accuracy: 0.9832
Epoch 3/10
20864/20864 [==============================] - 562s 27ms/step - loss: 0.0393 - crf_viterbi_accuracy: 0.9836 - val_loss: 0.0337 - val_crf_viterbi_accuracy: 0.9846
Epoch 4/10
20864/20864 [==============================] - 559s 27ms/step - loss: 0.0288 - crf_viterbi_accuracy: 0.9851 - val_loss: 0.0268 - val_crf_viterbi_accuracy: 0.9854
Epoch 5/10
20864/20864 [==============================] - 562s 27ms/step - loss: 0.0227 - crf_viterbi_accuracy: 0.9866 - val_loss: 0.0225 - val_crf_viterbi_accuracy: 0.9880
Epoch 6/10
20864/20864 [==============================] - 556s 27ms/step - loss: 0.0170 - crf_viterbi_accuracy: 0.9914 - val_loss: 0.0171 - val_crf_viterbi_accuracy: 0.9914
Epoch 7/10
20864/20864 [==============================] - 555s 27ms/step - loss: 0.0112 - crf_viterbi_accuracy: 0.9937 - val_loss: 0.0123 - val_crf_viterbi_accuracy: 0.9922
Epoch 8/10
20864/20864 [==============================] - 554s 27ms/step - loss: 0.0050 - crf_viterbi_accuracy: 0.9948 - val_loss: 0.0069 - val_crf_viterbi_accuracy: 0.9926
Epoch 9/10
20864/20864 [==============================] - 553s 26ms/step - loss: -0.0021 - crf_viterbi_accuracy: 0.9956 - val_loss: 2.2578e-04 - val_crf_viterbi_accuracy: 0.9930
Epoch 10/10
20864/20864 [==============================] - 551s 26ms/step - loss: -0.0097 - crf_viterbi_accuracy: 0.9962 - val_loss: -0.0067 - val_crf_viterbi_accuracy: 0.9931
              precision    recall  f1-score   support

         LOC     0.7336    0.6903    0.7113      3658
         ORG     0.7662    0.6343    0.6940      2185
         PER     0.8465    0.7720    0.8075      1864

   micro avg     0.7697    0.6942    0.7300      7707
   macro avg     0.7821    0.6989    0.7376      7707
weighted avg     0.7701    0.6942    0.7297      7707

**Comment by renkz** [comment]
你们的keras和tf版本是多少呀？ 我根本就跑不起来！
